<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Siming Zhao" />

<meta name="date" content="2024-01-07" />

<title>e1-GWAS-power</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">QBS-statsgen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/simingz/QBS-statsgen">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">e1-GWAS-power</h1>
<h4 class="author">Siming Zhao</h4>
<h4 class="date">2024-01-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>QBS-statsgen/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231230code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231230)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231230code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231230)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsimingzQBSstatsgentree3b4b60dd8ab1c2085c0aa825e4645f04833c1d9ctargetblank3b4b60da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/simingz/QBS-statsgen/tree/3b4b60dd8ab1c2085c0aa825e4645f04833c1d9c" target="_blank">3b4b60d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsimingzQBSstatsgentree3b4b60dd8ab1c2085c0aa825e4645f04833c1d9ctargetblank3b4b60da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/simingz/QBS-statsgen/tree/3b4b60dd8ab1c2085c0aa825e4645f04833c1d9c" target="_blank">3b4b60d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/B81CBE6F/bibliography-index/
    Ignored:    .Rproj.user/B81CBE6F/ctx/
    Ignored:    .Rproj.user/B81CBE6F/pcs/
    Ignored:    .Rproj.user/B81CBE6F/presentation/
    Ignored:    .Rproj.user/B81CBE6F/profiles-cache/
    Ignored:    .Rproj.user/B81CBE6F/sources/per/
    Ignored:    .Rproj.user/B81CBE6F/tutorial/
    Ignored:    .Rproj.user/shared/notebooks/1C2AC29C-e1-gwas-power/

Untracked files:
    Untracked:  .Rhistory
    Untracked:  analysis/e1-gwas-power.Rmd

Unstaged changes:
    Modified:   .Rproj.user/B81CBE6F/persistent-state
    Modified:   .Rproj.user/B81CBE6F/sources/prop/4C8B7780
    Modified:   .Rproj.user/B81CBE6F/sources/prop/BBFFB970
    Modified:   .Rproj.user/B81CBE6F/sources/prop/INDEX
    Deleted:    .Rproj.user/B81CBE6F/sources/s-e0e7218a/27031BD8
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/27031BD8-contents
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/34A40D3B
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/34A40D3B-contents
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/6C1FFABC
    Deleted:    .Rproj.user/B81CBE6F/sources/s-e0e7218a/73BDB580
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/lock_file
    Deleted:    .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/B81CBE6Fe0e7218a/chunks.json
    Modified:   .Rproj.user/shared/notebooks/paths
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="before-the-class-install-r-package-pwr-tidyverse"
class="section level2">
<h2>Before the class install R package <code>pwr</code>,
<code>tidyverse</code></h2>
<pre class="r"><code>install.packages(&quot;pwr&quot;)
install.packages(&quot;tidyverse&quot;)</code></pre>
</div>
<div id="workflow-overview" class="section level2">
<h2>Workflow overview</h2>
<p>In this vignette, we’d like to:</p>
<ol style="list-style-type: decimal">
<li>Write a simulator to simulate genotype and phenotype under some
pre-specified model.</li>
<li>Simulate genotype and phenotype using the simulator under the null
and alternative.</li>
<li>Perform certain hypothesis test.</li>
<li>Calculate the power of the test.</li>
</ol>
<p>Consider a single locus, its genotype is <span
class="math inline">\(X\)</span>. We pre-specify the model for
continuous trait <span class="math inline">\(Y\)</span> and genotype
<span class="math inline">\(X\)</span> as</p>
<p><span class="math display">\[Y = \beta X + \epsilon, \epsilon ~ N(0,
\sigma_\epsilon^2)\]</span> where we assume</p>
<ul>
<li>genotype of this locus follows Hardy-Weinberg equilibrium with a
pre-specified minor allele frequency.</li>
<li>trait is normalized to variance 1.</li>
</ul>
</div>
<div id="phenotype-genotype-simulator" class="section level2">
<h2>Phenotype-genotype simulator</h2>
<p>To simulate genotype, we assume the locus is bialleilic and each
individual is diploid. So that <span class="math inline">\(X \sim
Binomial(2, f)\)</span> with <span class="math inline">\(f\)</span> as
minor allele frequency (here we encode minor allele as 1 and major
allele as 0). with the linear model for simulate <span
class="math inline">\(Y\)</span> above, this basically means we are
simulating <span class="math inline">\(Y\)</span> under the additive
model.</p>
<p>Given genotype, to simulate phenotype, we need to know <span
class="math inline">\(\beta\)</span> and <span
class="math inline">\(\sigma_\epsilon^2\)</span>.</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.5     ✔ purrr   0.3.4
✔ tibble  3.1.2     ✔ dplyr   1.0.7
✔ tidyr   1.1.3     ✔ stringr 1.4.0
✔ readr   1.4.0     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>simulate_genotype = function(maf, num_individuals, num_replicates) {
  # maf: minor allele frequency
  # num_individuals: the number of individuals in each replicates
  # num_replicates: the number of replicates
  # it returns a matrix with num_individuals rows and num_replicates columns
  genotype = matrix( 
    rbinom(num_individuals * num_replicates, 2, maf), 
    nrow = num_individuals, 
    ncol = num_replicates 
  )
  return(genotype)
}
simulate_phenotype = function(genotype, beta, sig2epsi) {
  # genotype: each column is one replicate 
  # beta: effect size of the linear model
  # sig2epsi: the variance of the noise term
  num_individuals = nrow(genotype)
  num_replicates = ncol(genotype)
  epsilon = matrix( 
    rnorm(num_individuals * num_replicates, mean = 0, sd = sqrt(sig2epsi)), 
    nrow = num_individuals, 
    ncol = num_replicates 
  )
  phenotype = genotype * beta + epsilon
  return(phenotype)
}

linear_model_simulator = function(num_replicates, num_individuals, maf, beta, sig2epsi) {
  # simulate genotype
  X = simulate_genotype(maf, num_individuals, num_replicates)
  
  # simulate phenotype given genotype and model parameters
  Y = simulate_phenotype(X, beta, sig2epsi)
  return(list(Y = Y, X = X))
}</code></pre>
</div>
<div id="run-the-simulator-under-the-null-and-alternative"
class="section level2">
<h2>Run the simulator under the null and alternative</h2>
<p>Here we simulate 1000 individuals per replicate and 100 replicates in
total. With parameters:</p>
<ul>
<li>Minor allele frequency is 0.3.</li>
<li>effect size (<span class="math inline">\(\beta\)</span>) of the
minor allele 0.05. Effect size is the coefficient in a regression model
which measures the regression effect of the locus per copy of the
variant allele.</li>
<li>Variance of residual (<span
class="math inline">\(\sigma_\epsilon^2\)</span>) = 1.</li>
</ul>
<pre class="r"><code># specify paramters
nindiv = 1000
nreplicate = 5000
maf = 0.30
b = 0.05
sig2e = 1

# run simulator 
## under the alternative
data_alt = linear_model_simulator(nreplicate, nindiv, maf, 0.05, sig2e)  
## under the null
data_null = linear_model_simulator(nreplicate, nindiv, maf, 0, sig2e)  </code></pre>
</div>
<div id="perform-hypothesis-test" class="section level2">
<h2>Perform hypothesis test</h2>
<p>The following chunk of R code implement hypothesis test procedure
based on linear regression. Essentially, the R function
<code>calcz</code> takes genotype <code>X</code> and <code>Y</code> and
returns test statistic z-score.</p>
<pre class="r"><code>runassoc = function(X,Y)
{
  pvec = rep(NA,ncol(X))
  bvec = rep(NA,ncol(X))
  for(ss in 1:ncol(X))
  {
    x = X[,ss]
    y = Y[,ss]
    fit = lm(y~x)
    pvec[ss] = summary(fit)$coefficients[2,4]  
    bvec[ss] =  summary(fit)$coefficients[2,1]  
  }
  list(pvec=pvec, bvec=bvec)
}

p2z = function(b,p)
{
  ## calculate zscore from p-value and sign of effect size
  sign(b) * abs(qnorm(p/2))
}

calcz = function(X,Y)
{
  tempo = runassoc(X,Y)
  p2z(tempo$bvec,tempo$pvec)
}</code></pre>
<p>Now that we can calculate test statistics under the null and
alternative.</p>
<pre class="r"><code>Zalt = calcz(data_alt$X, data_alt$Y)
Znull = calcz(data_null$X, data_null$Y)

tibble(Y = c(Zalt,Znull), type=c(rep(&quot;alt&quot;,length(Zalt)),rep(&quot;null&quot;,length(Znull))) ) %&gt;% ggplot(aes(Y,fill=type)) + geom_density(color=NA,alpha=0.6) + theme_bw(base_size = 15)</code></pre>
<p><img src="figure/e1-gwas-power.Rmd/calc_test_statistic-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="calculate-power" class="section level2">
<h2>Calculate power</h2>
<pre class="r"><code>## define significance level

alpha = 0.01

## find threshold for rejection; we want P(Znull &gt; alpha/2) two-sided

threshold = quantile(Znull, 1 - alpha/2)

## calculate proportion of Zalt above threshold

mean(Zalt &gt; threshold)</code></pre>
<pre><code>[1] 0.048</code></pre>
</div>
<div id="check-with-pwr.r.test-function" class="section level2">
<h2>check with pwr.r.test function</h2>
<pre class="r"><code>library(pwr)
calc_r = function(b,maf,sdy) {sdx = sqrt(2 * maf * (1-maf)); sdx * b * sdy}
r = calc_r(b=b,maf=maf,sdy= sqrt(sig2e + b**2*2*maf*(1-maf)))
pwr.r.test(n = nindiv, r= r, sig.level = alpha)</code></pre>
<pre><code>
     approximate correlation power calculation (arctangh transformation) 

              n = 1000
              r = 0.03242071
      sig.level = 0.01
          power = 0.06048404
    alternative = two.sided</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /software/R-4.1.0-no-openblas-el7-x86_64/lib64/R/lib/libRblas.so
LAPACK: /software/R-4.1.0-no-openblas-el7-x86_64/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pwr_1.3-0       forcats_0.5.1   stringr_1.4.0   dplyr_1.0.7    
 [5] purrr_0.3.4     readr_1.4.0     tidyr_1.1.3     tibble_3.1.2   
 [9] ggplot2_3.3.5   tidyverse_1.3.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9       lubridate_1.7.10 assertthat_0.2.1 rprojroot_2.0.2 
 [5] digest_0.6.27    utf8_1.2.1       R6_2.5.0         cellranger_1.1.0
 [9] backports_1.2.1  reprex_2.0.0     evaluate_0.20    highr_0.9       
[13] httr_1.4.2       pillar_1.6.1     rlang_1.1.0      readxl_1.3.1    
[17] rstudioapi_0.13  jquerylib_0.1.4  rmarkdown_2.21   labeling_0.4.2  
[21] munsell_0.5.0    broom_0.7.8      compiler_4.1.0   httpuv_1.6.1    
[25] modelr_0.1.8     xfun_0.38        pkgconfig_2.0.3  htmltools_0.5.5 
[29] tidyselect_1.1.1 workflowr_1.6.2  fansi_0.5.0      crayon_1.5.2    
[33] dbplyr_2.1.1     withr_2.5.0      later_1.2.0      grid_4.1.0      
[37] jsonlite_1.7.2   gtable_0.3.0     lifecycle_1.0.3  DBI_1.1.1       
[41] git2r_0.28.0     magrittr_2.0.1   scales_1.1.1     cli_3.6.1       
[45] stringi_1.6.2    cachem_1.0.5     farver_2.1.0     fs_1.6.1        
[49] promises_1.2.0.1 xml2_1.3.2       bslib_0.4.2      ellipsis_0.3.2  
[53] generics_0.1.0   vctrs_0.3.8      tools_4.1.0      glue_1.4.2      
[57] hms_1.1.0        fastmap_1.1.0    yaml_2.2.1       colorspace_2.0-2
[61] rvest_1.0.0      knitr_1.42       haven_2.4.1      sass_0.4.0      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
