<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Siming Zhao" />

<meta name="date" content="2023-12-31" />

<title>Exercise-1-GWAS</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">QBS-statsgen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/simingz/QBS-statsgen">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Exercise-1-GWAS</h1>
<h4 class="author">Siming Zhao</h4>
<h4 class="date">2023-12-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>QBS-statsgen/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231230code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231230)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231230code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231230)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsimingzQBSstatsgentreee4808cde19af6af5cea539b5d0bf5c358d626cb2targetblanke4808cda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/simingz/QBS-statsgen/tree/e4808cde19af6af5cea539b5d0bf5c358d626cb2" target="_blank">e4808cd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsimingzQBSstatsgentreee4808cde19af6af5cea539b5d0bf5c358d626cb2targetblanke4808cda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/simingz/QBS-statsgen/tree/e4808cde19af6af5cea539b5d0bf5c358d626cb2" target="_blank">e4808cd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/B81CBE6F/bibliography-index/
    Ignored:    .Rproj.user/B81CBE6F/ctx/
    Ignored:    .Rproj.user/B81CBE6F/presentation/
    Ignored:    .Rproj.user/B81CBE6F/profiles-cache/
    Ignored:    .Rproj.user/B81CBE6F/tutorial/

Untracked files:
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/0ED98C8D
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/9AEFAC31
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/9F0ECC46
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/C51E8536
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/CEEC26AB
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/D152DC47
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/EC99B653
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/F22062D7
    Untracked:  .Rproj.user/B81CBE6F/sources/prop/FB5E64F1
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/2CC61BD1-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/41809F0E
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/41809F0E-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/4606F1D0
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/4606F1D0-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/67181385-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/690E4D46-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/694B620B
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/694B620B-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/6B86B775-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/7DAD17BD-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/B673F567
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/B673F567-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/C175B722-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/D6419CAC
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/D6419CAC-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/D7E31414-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/E34D4A4D-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/EDA99CEB-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/F254ABAC-contents
    Untracked:  .Rproj.user/B81CBE6F/sources/s-e0e7218a/FA39596B-contents
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/c1lzammkxrbye/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/c1t2nsjfm2qe6/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/c253kmjzy3wx6/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/c33ie2g7kdxi0/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/cg1sv066q5yme/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/cgshy06jz7z7m/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/cjf82vg1kx5mw/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/cllcn96htab67/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/clxqgcy429ntx/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/cn50h0v0w0zuw/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/csdd4y3yd95j4/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/cxwhmvf97sdcw/
    Untracked:  .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/cybrmunyg5sbi/
    Untracked:  .gitignore
    Untracked:  code/qqunif.R
    Untracked:  output/asso.assoc.logistic
    Untracked:  output/asso.log
    Untracked:  output/pca.eigenval
    Untracked:  output/pca.eigenvec
    Untracked:  output/pca.log

Unstaged changes:
    Modified:   .Rproj.user/B81CBE6F/persistent-state
    Modified:   .Rproj.user/B81CBE6F/sources/prop/3814FE49
    Modified:   .Rproj.user/B81CBE6F/sources/prop/4C8B7780
    Modified:   .Rproj.user/B81CBE6F/sources/prop/51332097
    Modified:   .Rproj.user/B81CBE6F/sources/prop/BBFFB970
    Modified:   .Rproj.user/B81CBE6F/sources/prop/INDEX
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/27031BD8
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/27031BD8-contents
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/34A40D3B
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/6C1FFABC
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/6C1FFABC-contents
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/73BDB580
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/73BDB580-contents
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/lock_file
    Modified:   .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/B81CBE6Fe0e7218a/chunks.json
    Modified:   .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/chunks.json
    Modified:   .Rproj.user/shared/notebooks/paths
    Modified:   _workflowr.yml
    Modified:   analysis/_site.yml
    Modified:   analysis/e1-gwas.Rmd
    Deleted:    output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/e1-gwas.Rmd</code>) and HTML
(<code>docs/e1-gwas.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/QBS-statsgen/blob/e4808cde19af6af5cea539b5d0bf5c358d626cb2/analysis/e1-gwas.Rmd" target="_blank">e4808cd</a>
</td>
<td>
simingz
</td>
<td>
2023-12-31
</td>
<td>
build website
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="before-the-class" class="section level1">
<h1>Before the class</h1>
<ul>
<li>Install <code>plink</code>.</li>
</ul>
<p>Two versions</p>
<ul>
<li><p><a href="https://www.cog-genomics.org/plink2">PLINK Homepage
(1.9)</a></p></li>
<li><p><a href="https://www.cog-genomics.org/plink2.0">PLINK Homepage
(2.0)</a></p></li>
</ul>
<p>For this exercise, please install plink 1.9. On discovery, plink is
already installed. you can do <code>module load plink</code> to load
plink.</p>
<ul>
<li><p>Install R &gt;= 4.1, Rstudio. You can do this on your local
computer, the discovery cluster, or use <a href="https://posit.cloud/"
class="uri">https://posit.cloud/</a>. The free plan offers 25
hours/month.</p></li>
<li><p>Install R packages, <code>snpStats</code>,
<code>data.table</code>, <code>glue</code>, <code>dplyr</code>,
<code>qqman</code></p></li>
</ul>
<pre class="r"><code>install.packages(&#39;BiocManager&#39;)
BiocManager::install(&#39;snpStats&#39;)
install.packages(&quot;data.table&quot;) 
install.packages(&quot;dplyr&quot;)
install.packages(&quot;qqman&quot;)</code></pre>
<ul>
<li>Getting the data</li>
</ul>
<p>This tutorial will use data from the PennCATH study of genetic risk
factors for coronary artery disease. Download the data from one of the
following sources (the contents are the same):</p>
<ul>
<li><a
href="https://d1ypx1ckp5bo16.cloudfront.net/penncath/penncath.zip">Data
(zip)</a>: Windows</li>
<li><a
href="https://d1ypx1ckp5bo16.cloudfront.net/penncath/penncath.tar.gz">Data
(tar)</a>: Mac/Linux</li>
</ul>
<p>Download and unzip/untar the data; you can read the paper as well if
you wish:</p>
<ul>
<li><a href="http://www.ncbi.nlm.nih.gov/pubmed/21239051">Paper</a></li>
</ul>
<p>The process this data for this tutorial (the orginal ones have a
split chromosome issue)</p>
<p>In command line run this:</p>
<pre class="bash"><code>plink -bfile data/penncath --make-bed --out data/geno</code></pre>
<p>If your plink is installed elsewhere, e.g. in <code>/bin</code> then
you need to run it like <code>/bin/plink</code>. On discovery, plink is
already installed. you can do <code>module load plink</code> to load
plink then run the command above.</p>
</div>
<div id="file-formats" class="section level1">
<h1>File formats</h1>
<p>The data are given in “PLINK” format, which is the most common format
for chip-based GWAS data (as of this writing!). <code>PLINK</code> is an
open-source whole genome association analysis toolset designed to
perform a range of basic large-scale analyses in a computationally
efficient manner. It is worth knowing how to use PLINK, although you can
also do most of these things in R.</p>
<p>I’ll discuss PLINK the software program later on; for now, I’ll just
describe the organization of its files.</p>
<p>Among the zipped files are three that are necessary to perform a
GWAS, the <code>.bed</code>, <code>.bim</code>, and <code>.fam</code>
files.</p>
<div id="fam" class="section level2">
<h2><code>.fam</code></h2>
<p>This contains information on the subjects:</p>
<pre class="r"><code>library(data.table)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>(fam &lt;- fread(&#39;data/geno.fam&#39;))</code></pre>
<pre><code>         V1 V2 V3 V4 V5 V6
   1: 10002  1  0  0  1  1
   2: 10004  1  0  0  2  1
   3: 10005  1  0  0  1  2
   4: 10007  1  0  0  1  1
   5: 10008  1  0  0  1  2
  ---                     
1397: 11591  1  0  0  2 -9
1398: 11592  1  0  0  1  2
1399: 11593  1  0  0  1  1
1400: 11594  1  0  0  1  1
1401: 11596  1  0  0  1 -9</code></pre>
<p>There are 1401 rows, one for each subject. The six colums are:</p>
<ol style="list-style-type: decimal">
<li>Family ID</li>
<li>Individual ID</li>
<li>Paternal ID</li>
<li>Maternal ID</li>
<li>Sex (1=male; 2=female; other=unknown)</li>
<li>Phenotype</li>
</ol>
<p>In this data set, columns 2-4 are unimportant. In general, they are
used to specify pedigrees (e.g., subject 3 is the daughter of subjects 1
and 2). In this study, however, none of the subjects are related, so the
only column that is important is the first, which records the subject’s
unique ID.</p>
<p>Phenotype is typically used to record case-control status or
something like that, but it is also quite common to just record
clinical/biological information in a separate spreadsheet, which is what
was done here.</p>
<pre class="r"><code>(clinical &lt;- fread(&#39;data/penncath.csv&#39;))</code></pre>
<pre><code>      FamID CAD sex age  tg hdl ldl
   1: 10002   1   1  60  NA  NA  NA
   2: 10004   1   2  50  55  23  75
   3: 10005   1   1  55 105  37  69
   4: 10007   1   1  52 314  54 108
   5: 10008   1   1  58 161  40  94
  ---                              
1397: 11591   0   2  59  34  44  89
1398: 11592   1   1  45  69 101  77
1399: 11593   1   1  59  77  27  41
1400: 11594   1   1  30  NA  NA  NA
1401: 11596   0   1  64 224  35  96</code></pre>
<p>As you can see, we’ve got the <code>FamID</code> to match this
spreadsheet up with the genetic data, the disease status
(<code>CAD=1</code> means that the subject has coronary artery disease),
and some covariates (age, triglycerides, HDL and LDL cholesterol
levels).</p>
</div>
<div id="bim" class="section level2">
<h2><code>.bim</code></h2>
<p>The <code>.bim</code> file, by contrast, contains information on the
genetic loci (SNPs):</p>
<pre class="r"><code>(bim &lt;- fread(&#39;data/geno.bim&#39;))</code></pre>
<pre><code>        V1         V2 V3       V4 V5 V6
     1:  1 rs10458597  0   564621 -9  C
     2:  1 rs12565286  0   721290  G  C
     3:  1 rs12082473  0   740857  T  C
     4:  1  rs3094315  0   752566  C  T
     5:  1  rs2286139  0   761732  C  T
    ---                                
861469: 22  rs3810648  0 51175626  C  T
861470: 22  rs3865766  0 51186228  A  G
861471: 22  rs2238837  0 51212875  C  A
861472: 22 rs34726907  0 51213613  T  C
861473: 22 rs28729663  0 51219006  A  G</code></pre>
<p>As you can see, we have 861473 rows here, one for each SNP measured
in the study. The columns are:</p>
<ol style="list-style-type: decimal">
<li>chromosome (1-22, X, Y or 0 if unplaced)</li>
<li>rs# or snp identifier</li>
<li>Genetic distance (morgans)</li>
<li>Base-pair position (bp units)</li>
<li>Allele 1 (usually minor)</li>
<li>Allele 2 (usually major)</li>
</ol>
<p>It is pretty common for column 3 to be ignored, as it is here.</p>
<p>So, for example, the file tells us that genetic locus rs12565286 is
located 721290 bases into chromosome 1, and that most people have a C
there, but some have a G.</p>
</div>
<div id="bed" class="section level2">
<h2><code>.bed</code></h2>
<p>Finally, the <code>.bed</code> file, which has all the data. This is
by far the largest of the three files, as it contains the entire 1401 by
861473 matrix of genotype calls for every subject and every locus. To
keep things manageable, this file is encoded in a special binary format
– i.e., you can’t just read it in through normal means.</p>
<p>To access it, you’ll have to use specialized applications. I’ll
discuss two, an R package called <code>snpStats</code> and a
command-line interface (CLI) called PLINK.</p>
</div>
</div>
<div id="software" class="section level1">
<h1>Software</h1>
<div id="plink" class="section level2">
<h2>plink</h2>
<p>How to convert your genetic data into plink format?</p>
<p>One can use plink2 binary software to convert your genetic dataset
into this format. For example, the following –make-pgen (link to
reference) command will convert your input.vcf file into
binary_fileset.{pgen,pvar,psam}. Many other input formats are also
supported in plink2.</p>
<pre class="r"><code>plink2 --vcf input.vcf --make-pgen --out binary_fileset</code></pre>
<p>Similarly, you can convert your genetic dataset into plink1.9 binary
format (i.e. the triple of bed,bim,fam) with –make-bed (link to
reference) command.</p>
<pre class="r"><code>plink2 --vcf input.vcf --make-bed --out binary_fileset</code></pre>
<p>Note that the file extension bed here refers to the binary PED
format. It is nothing to do with UCSC’s BED (Browser Extensible Data)
file format.</p>
</div>
<div id="snpstats" class="section level2">
<h2>snpStats</h2>
<p>This is a Bioconductor package. So, you’ll have to install it via
<code>BiocManager</code></p>
<pre class="r"><code>library(snpStats)</code></pre>
<pre><code>Loading required package: survival</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<p>To read in data, there is the <code>read.plink()</code> function:</p>
<pre class="r"><code>obj &lt;- read.plink(&#39;data/geno&#39;)</code></pre>
<p>The function assumes that all the files have the same base filename,
and differ only in their extension. If this is not the case, then you
need to specify the filenames for the <code>.bed</code>,
<code>.bim</code>, and <code>.fam</code> files separately.</p>
<p>From here, <code>snpStats</code> has a lot of functions. For example,
here’s a plot (there are 1401 points, one for each subject) of whether
the call rate (% of genotype calls that are non-missing) is related to
the heterozygosity rate (% of loci that are called AB, as opposed to AA
or BB):</p>
<pre class="r"><code>plot(row.summary(obj$genotypes)[c(1,3)])</code></pre>
<p><img src="figure/e1-gwas.Rmd/snp-summary-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Feel free to read the <code>snpStats</code> documentation and explore
for yourself, but one standard thing that one is always interested in is
to simply convert various SNPs to a regular numeric matrix so that you
can analyze them using standard R tools. For example, let’s do a
Fisher’s exact test to see whether CAD is associated with SNP 143:</p>
<pre class="r"><code>x &lt;- as(obj$genotypes[,143], &#39;numeric&#39;)
fisher.test(drop(x), as.factor(clinical$CAD))</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  drop(x) and as.factor(clinical$CAD)
p-value = 0.8043
alternative hypothesis: two.sided</code></pre>
</div>
<div id="other-r-packages-to-read-plink-files." class="section level2">
<h2>Other R packages to read plink files.</h2>
<p>There are several other R packages that can work with PLINK formatted
files. - <a href="https://github.com/gabraham/plink2R">plink2R</a> - <a
href="https://github.com/chrchang/plink-ng">pgenlibr</a>, this package
can work with plink2 formatted files, ie .pgen, pvar, psam. - <a
href="https://privefl.github.io/bigsnpr/">bigsnpr</a></p>
</div>
</div>
<div id="gwas-quality-control-qc" class="section level1">
<h1>GWAS Quality Control (QC)</h1>
<p>Check a few things. In actual GWAS, there are more to check.</p>
<p>Any SNP with a lot of missing data is probably questionable; these
SNPs are often excluded from analysis (although we will talk about other
approaches later). Likewise, any sample with lots of missing data
suggests that there may be issues with the processing of that
sample.</p>
<pre class="r"><code>cs &lt;- col.summary(obj$genotypes)
rs &lt;- row.summary(obj$genotypes)

hist(rs$Call.rate, main = &#39;Individuals Call rate&#39;)
hist(cs$Call.rate, main = &#39;SNPs Call rate&#39;)</code></pre>
<p><img src="figure/e1-gwas.Rmd/miss-1.png" width="50%" /><img src="figure/e1-gwas.Rmd/miss-2.png" width="50%" /></p>
<p>Individuals look good – SNPs, on the other hand, there are definitely
some SNPs with lots of missing values. A common practice is to exclude
SNPs with &gt;5% or &gt;10% missing data.</p>
<p>We will remove SNPs that may be problematic.</p>
<pre class="r"><code>p_hwe &lt;- 2*pnorm(-abs(cs$z.HWE)) # hardy-weinberg equilibrium. extreme value may indicate genotype error.

keep &lt;- cs$MAF &gt; 0.001 &amp;
  cs$Call.rate &gt; 0.9 &amp; 
   abs(cs$z.HWE) &lt; 6.5

(obj$genotypes &lt;- obj$genotypes[, keep])</code></pre>
<pre><code>A SnpMatrix with  1401 rows and  752675 columns
Row names:  10002 ... 11596 
Col names:  rs12565286 ... rs28729663 </code></pre>
<pre class="r"><code>obj$map &lt;- obj$map[keep, ]

# write plnik files for filtered data.
write.plink(
  file.base = &quot;data/geno.qc&quot;,
  snps = obj$genotypes,
  pedigree = obj$fam$pedigree,
  id = obj$fam$member,
  father = obj$fam$father,
  mother = obj$fam$mother,
  chromosome = obj$map$chromosome,
  genetic.distance = obj$map$cM,
  position = obj$map$position,
  allele.1 = obj$map$allele.1,
  allele.2 = obj$map$allele.2,
  sex = clinical$sex,
  phenotype = clinical$CAD + 1
)</code></pre>
<pre><code>Writing FAM file to data/geno.qc.fam 
Writing extended MAP file to data/geno.qc.bim 
Writing BED file to data/geno.qc.bed (SNP-major mode)</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="population-structure." class="section level1">
<h1>Population structure.</h1>
<p>It is possible to run PCA inside R, but given the large file size.
Usually this is done by running plink from the command line.</p>
<pre class="bash"><code>plink -bfile data/geno.qc --pca --out output/pca</code></pre>
<p>Then in R, we read in PCs and plot.</p>
<pre class="r"><code>## read plink calculated PCs
pcplink = read.table(&quot;output/pca.eigenvec&quot;,header=F, as.is=T)
names(pcplink) = c(&quot;FID&quot;,&quot;IID&quot;,paste0(&quot;PC&quot;, c(1:(ncol(pcplink)-2))) )

## plot PC1 vs PC2
plot(pcplink$PC1, pcplink$PC2)</code></pre>
<p><img src="figure/e1-gwas.Rmd/PCA-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot variance explained by PCs
eigen = read.table(&quot;output/pca.eigenval&quot;,header=F, as.is=T)[,1]
plot(1:length(eigen), eigen/nrow(pcplink), type = &#39;b&#39;, ylab = &#39;Proportion of variance explained&#39;, xlab = &#39;PC&#39;)</code></pre>
<p><img src="figure/e1-gwas.Rmd/PCA-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="association-testing" class="section level1">
<h1>Association testing</h1>
<p>Run logistic regression using plink, adjusting for PCs. Run the
following from command line.</p>
<pre class="bash"><code>plink --bfile data/geno.qc --logistic --pheno data/penncath.csv --pheno-name CAD --covar output/pca.eigenvec --covar-number 1-4 --out output/asso</code></pre>
<p>Read results into R and plot</p>
<pre class="r"><code>res &lt;- fread(&#39;output/asso.assoc.logistic&#39;)
res &lt;- res[res$TEST == &quot;ADD&quot;,]
source(&quot;code/qqunif.R&quot;)
qqunif(res$P, mlog10_p_thres=12)</code></pre>
<pre><code>Warning in qqunif(res$P, mlog10_p_thres = 12): thresholding p to 1e-12</code></pre>
<p><img src="figure/e1-gwas.Rmd/plotres-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Manhattan plot.</p>
<pre class="r"><code>bim &lt;- fread(&#39;data/geno.qc.bim&#39;)
# format data to have readable labels. 
manh_data &lt;- data.frame(
    SNP = res$SNP,
    P = res$P
  ) %&gt;%
  left_join(bim, by = c(&quot;SNP&quot; = &quot;V2&quot;)) %&gt;%
  rename(
    CHR = V1, 
    BP = V4
  ) # recall that &#39;bim&#39; files have a standardized format, so the column order is 
# always the same 
library(qqman)</code></pre>
<pre><code></code></pre>
<pre><code>For example usage please run: vignette(&#39;qqman&#39;)</code></pre>
<pre><code></code></pre>
<pre><code>Citation appreciated but not required:</code></pre>
<pre><code>Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. Journal of Open Source Software, 3(25), 731, https://doi.org/10.21105/joss.00731.</code></pre>
<pre><code></code></pre>
<pre class="r"><code>manh_plot &lt;- manhattan(manh_data, ylim = c(0, 10), annotatePval = 5e-6)</code></pre>
<p><img src="figure/e1-gwas.Rmd/plotmanh-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="citation" class="section level1">
<h1>Citation</h1>
<p>Credits go to: <a
href="https://github.com/pbreheny/adv-gwas-tutorial/tree/master"
class="uri">https://github.com/pbreheny/adv-gwas-tutorial/tree/master</a>.<br />
<a href="https://yosuketanigawa.com/posts/2020/09/PLINK2"
class="uri">https://yosuketanigawa.com/posts/2020/09/PLINK2</a><br />
<a
href="https://hakyimlab.github.io/hgen471/L6-population-structure.html"
class="uri">https://hakyimlab.github.io/hgen471/L6-population-structure.html</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /software/R-4.1.0-no-openblas-el7-x86_64/lib64/R/lib/libRblas.so
LAPACK: /software/R-4.1.0-no-openblas-el7-x86_64/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] qqman_0.1.9       snpStats_1.44.0   Matrix_1.3-3      survival_3.2-11  
[5] dplyr_1.0.7       data.table_1.14.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.1    xfun_0.38           bslib_0.4.2        
 [4] purrr_0.3.4         splines_4.1.0       lattice_0.20-44    
 [7] vctrs_0.3.8         generics_0.1.0      htmltools_0.5.5    
[10] yaml_2.2.1          utf8_1.2.1          rlang_1.1.0        
[13] jquerylib_0.1.4     later_1.2.0         pillar_1.6.1       
[16] glue_1.4.2          DBI_1.1.1           BiocGenerics_0.40.0
[19] calibrate_1.7.7     lifecycle_1.0.3     stringr_1.4.0      
[22] zlibbioc_1.40.0     workflowr_1.6.2     evaluate_0.20      
[25] knitr_1.42          fastmap_1.1.0       httpuv_1.6.1       
[28] fansi_0.5.0         highr_0.9           Rcpp_1.0.9         
[31] promises_1.2.0.1    cachem_1.0.5        jsonlite_1.7.2     
[34] fs_1.6.1            digest_0.6.27       stringi_1.6.2      
[37] rprojroot_2.0.2     grid_4.1.0          cli_3.6.1          
[40] tools_4.1.0         magrittr_2.0.1      sass_0.4.0         
[43] tibble_3.1.2        crayon_1.5.2        whisker_0.4        
[46] pkgconfig_2.0.3     MASS_7.3-54         ellipsis_0.3.2     
[49] assertthat_0.2.1    rmarkdown_2.21      rstudioapi_0.13    
[52] R6_2.5.0            git2r_0.28.0        compiler_4.1.0     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
