<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Siming Zhao" />

<meta name="date" content="2024-01-21" />

<title>Polygenic risk score</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">QBS-statsgen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/simingz/QBS-statsgen">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Polygenic risk score</h1>
<h4 class="author">Siming Zhao</h4>
<h4 class="date">2024-01-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>QBS-statsgen/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231230code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231230)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231230code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231230)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsimingzQBSstatsgentree85e59dd679229bf0c05809786a48f5a4a18c9e9btargetblank85e59dda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/simingz/QBS-statsgen/tree/85e59dd679229bf0c05809786a48f5a4a18c9e9b" target="_blank">85e59dd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsimingzQBSstatsgentree85e59dd679229bf0c05809786a48f5a4a18c9e9btargetblank85e59dda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/simingz/QBS-statsgen/tree/85e59dd679229bf0c05809786a48f5a4a18c9e9b" target="_blank">85e59dd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/B81CBE6F/bibliography-index/
    Ignored:    .Rproj.user/B81CBE6F/ctx/
    Ignored:    .Rproj.user/B81CBE6F/pcs/
    Ignored:    .Rproj.user/B81CBE6F/presentation/
    Ignored:    .Rproj.user/B81CBE6F/profiles-cache/
    Ignored:    .Rproj.user/B81CBE6F/sources/per/
    Ignored:    .Rproj.user/B81CBE6F/tutorial/
    Ignored:    .Rproj.user/shared/notebooks/1C2AC29C-e1-gwas-power/
    Ignored:    .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/ce0r78nx8keuu/
    Ignored:    .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/csetup_chunk/
    Ignored:    .Rproj.user/shared/notebooks/1EB0B2DC-e1-gwas/1/s/czxn6jf8lsykc/
    Ignored:    .Rproj.user/shared/notebooks/26ED8139-e2-prs/
    Ignored:    .Rproj.user/shared/notebooks/BC66D613-e2-lmm/
    Ignored:    .Rproj.user/shared/notebooks/FCFC3BD0-e2-finemapping/
    Ignored:    data/e2-ori/
    Ignored:    data/e2/
    Ignored:    output/

Untracked files:
    Untracked:  analysis/e2-finemapping.Rmd
    Untracked:  analysis/e2-lmm.Rmd
    Untracked:  analysis/e2-prs.Rmd
    Untracked:  code/Bayesian-linear-regression.R

Unstaged changes:
    Modified:   .Rproj.user/B81CBE6F/persistent-state
    Modified:   .Rproj.user/B81CBE6F/sources/prop/4C8B7780
    Modified:   .Rproj.user/B81CBE6F/sources/prop/BBFFB970
    Modified:   .Rproj.user/B81CBE6F/sources/prop/INDEX
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/34A40D3B
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/34A40D3B-contents
    Deleted:    .Rproj.user/B81CBE6F/sources/s-e0e7218a/6C1FFABC
    Modified:   .Rproj.user/B81CBE6F/sources/s-e0e7218a/lock_file
    Modified:   .Rproj.user/shared/notebooks/paths
    Modified:   analysis/index.Rmd
    Deleted:    temp.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="before-the-class" class="section level2">
<h2>Before the Class</h2>
<ul>
<li>Install GCTA</li>
</ul>
<p>The GCTA software can be downloaded from <a
href="https://yanglab.westlake.edu.cn/software/gcta/#Download"
class="uri">https://yanglab.westlake.edu.cn/software/gcta/#Download</a></p>
<p>This is the original paper for GCTA: Yang et al. (2011) GCTA: a tool
for Genome-wide Complex Trait Analysis. Am J Hum Genet. 88(1): 76-82.
[PubMed ID: 21167468]. This is a tool widely used to estimate and
partition complex trait variation with large GWAS data sets.</p>
<ul>
<li>Download data</li>
</ul>
<p>From here <a
href="https://rcweb.dartmouth.edu/Szhao/QBS148-statsgen/e2/"
class="uri">https://rcweb.dartmouth.edu/Szhao/QBS148-statsgen/e2/</a>.</p>
</div>
<div id="data-used-in-this-analysis" class="section level2">
<h2>Data used in this analysis</h2>
<ul>
<li>Genotype data The genotype data is given plink1 format:
<code>oz.fam</code>, <code>oz.bim</code>, <code>oz.bed</code></li>
</ul>
<p>Pay attention to the .fam file</p>
<pre class="bash"><code>head data/e2/oz.fam</code></pre>
<pre><code>1400 1400t1 0 0 1 -9
1400 1400t2 0 0 2 -9
570 570t1 0 0 1 -9
570 570t2 0 0 1 -9
3413 3413t1 0 0 2 -9
3413 3413t2 0 0 1 -9
1911 1911t1 0 0 2 -9
1911 1911t2 0 0 2 -9
1403 1403t1 0 0 2 -9
1403 1403t2 0 0 1 -9</code></pre>
<p>The .fam file has 6 columns: Family ID, Individual ID, Fathers ID
(0=missing), Mothers ID (0=missing), Sex (1=M, 2=F), Phenotype
(-9=missing).</p>
<ul>
<li>Phenotype data</li>
</ul>
<pre class="bash"><code>head data/e2/ozht.phen</code></pre>
<pre><code>FID IID ht
1000 1000t1 167.99
1000 1000t2 167.99
1001 1001t1 173
1001 1001t2 164.99
1002 1002t1 164.99
1002 1002t2 151.98
1004 1004t1 162.99
1004 1004t2 156.98
1005 1005t1 176.98</code></pre>
<p>This file contains 3 columns: Family ID, Individual ID, Height (in
cm)</p>
<ul>
<li>Covariates file</li>
</ul>
<pre class="bash"><code>head data/e2/ozht.covar
head data/e2/ozht.qcovar</code></pre>
<pre><code>FID IID sex
1000 1000t1 2
1000 1000t2 2
1001 1001t1 2
1001 1001t2 2
1002 1002t1 2
1002 1002t2 2
1004 1004t1 2
1004 1004t2 2
1005 1005t1 2
FID IID age PC1 PC2 PC3 PC4
1000 1000t1 23 0.0105 0.0279 -0.0088 -0.0156
1000 1000t2 23 0.0106 0.0267 -0.0101 -0.0161
1001 1001t1 17 0.0101 0.0259 -0.0089 -0.0163
1001 1001t2 17 0.0104 0.0266 -0.0096 -0.0171
1002 1002t1 20 0.0109 0.0269 -0.0104 -0.0188
1002 1002t2 20 0.0105 0.027 -0.0101 -0.0172
1004 1004t1 19 0.0084 0.0258 -0.007 -0.0153
1004 1004t2 19 0.0089 0.0256 -0.0045 -0.0159
1005 1005t1 20 0.0105 0.0256 -0.0097 -0.0209</code></pre>
<p>We have two covariates file, one for categorical covariates
<code>ozht.covar</code> This file contains 8 columns: Family ID,
Individual ID, Age, Sex (1=M,2=F), and 4 genetic principle components
which we will use to account for the effects of ethnicity in our
analyses.</p>
<ul>
<li>Summary statistics file</li>
</ul>
<pre class="bash"><code>head data/e2/weights.prs</code></pre>
<pre><code>SNP CHR BP A1 A2 BETA P
SNP6438 1 995985 T C 9.22365e-03 0.000126
SNP2310 1 1299528 C A 1.32753e-02 9.11e-08
SNP3090 1 1483355 A G 9.02390e-03 3.58e-07
SNP10481 1 2065231 T G 1.02686e-02 0.0113
SNP2707 1 2118624 T C 9.23841e-03 1.93e-07
SNP886 1 2142211 C A 1.63796e-02 1.24e-18
SNP2202 1 2248368 T C 1.49619e-02 7.52e-08
SNP10484 1 2248297 C A -4.84493e-03 0.0113
SNP3705 1 2274422 G A -9.42793e-03 8.48e-07</code></pre>
<p>In this file, A1 is the effect allele.</p>
</div>
<div id="compute-prs-using-the-cpt-strategy." class="section level2">
<h2>Compute PRS using the C+PT strategy.</h2>
<p>The first method is classically denoted as “Clumping + P-value
Thresholding (C+PT)”. This method is also abbreviated as P+T or C+T in
certain publications. In brief, the principle of this method is compare
various sets of uncorrelated SNPs (e.g., maximum squared pairwise
correlation between allele counts at SNPs in the selected set is r2≤0.1
) that are associated with the trait / disease as certain p-value
threshold (e.g., p&lt;0.01). We then select the set of SNPs that yields
the largest prediction accuracy with the trait / disease of interest in
a validation set. This method is broadly used because of it simplicity
but may not often yield the largest accuracy. In this practical, we will
use PLINK and R to determine these optimal sets of SNPs. PGS will be
calculated using marginal/GWAS SNP effects as weights.</p>
<p><strong>Step 1</strong>: clump and select SNPs. Run from command
line:</p>
<pre class="bash"><code>window_kb=1000 # 1000 kb = 1 Mb window
r2_thresh=0.1  # LD threshold for clumping
pv_thresh=5e-8

plink --bfile data/e2/oz \
      --clump data/e2/weights.prs \
      --clump-kb ${window_kb} \
      --clump-p1 ${pv_thresh} \
      --clump-p2 ${pv_thresh} \
      --clump-r2 ${r2_thresh} \
      --out output/c-pt_rsq_${r2_thresh}_p_below_${pv_thresh}</code></pre>
<p>How many SNPs are picked?</p>
<p><strong>Step 2</strong>: calculate the PGS with PLINK using the
–score command (help: <a
href="https://www.cog-genomics.org/plink/1.9/score"
class="uri">https://www.cog-genomics.org/plink/1.9/score</a>).</p>
<pre class="bash"><code>plink --bfile data/e2/oz\
      --score data/e2/weights.prs 1 4 6 header sum \
      --extract output/c-pt_rsq_${r2_thresh}_p_below_${pv_thresh}.clumped \
      --out output/c-pt_rsq_${r2_thresh}_p_below_${pv_thresh}.pred</code></pre>
<p>Assuming that the 1rt column is the SNP ID; 4th column is the
effective allele information; the 6th column is the effect size
estimate; and that the file contains a header. We use Sum of the
effects. See <a
href="https://www.cog-genomics.org/plink/1.9/score">here</a> for plink
documentation.</p>
<p>Take a look at the output:</p>
<pre class="bash"><code>head output/c-pt_rsq_0.1_p_below_5e-8.pred.profile</code></pre>
<pre><code>    FID      IID  PHENO    CNT   CNT2 SCORESUM
   1400   1400t1     -9   1902    353 -0.38628
   1400   1400t2     -9   1786    340 0.222768
    570    570t1     -9   1770    355 -0.208286
    570    570t2     -9   1770    355 -0.208286
   3413   3413t1     -9   1686    341 -0.0885919
   3413   3413t2     -9   1682    320 -0.199319
   1911   1911t1     -9   1970    399 -0.336112
   1911   1911t2     -9   1970    399 -0.336112
   1403   1403t1     -9   1774    362 -0.576168</code></pre>
<p>We will use linear mixed model to assess r2 of our PRS score as our
samples contains related individuals. (If all individuals are unrelated,
we can just use linear regression.) First prepare covariates file, the
last column is the PRS score</p>
<pre class="r"><code>qcovar &lt;- read.table(&quot;data/e2/ozht.qcovar&quot;, header = T)
prs &lt;- read.table(&quot;output/c-pt_rsq_0.1_p_below_5e-8.pred.profile&quot;, header = T)
all=merge(qcovar, prs,  by=c(&quot;IID&quot;, &quot;FID&quot;))
write.table(all[,c(&quot;FID&quot;, &quot;IID&quot;, &quot;age&quot;, &quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;SCORESUM&quot;)], &quot;output/ozht.prs_rsq_0.1_p_below_5e-8.qcovar&quot;, quote = F, row.names = F)</code></pre>
<p>Then run GCTA to estimate fixed effect size for PRS score.</p>
<p>From command line run this:</p>
<pre class="bash"><code>gcta --bfile data/e2/oz --make-grm --out output/ozGCTA
gcta --reml\
     --reml-est-fix\
     --grm output/ozGCTA\
     --pheno data/e2/ozht.phen\
     --qcovar output/ozht.prs_rsq_${r2_thresh}_p_below_${pv_thresh}.qcovar\
     --covar data/e2/ozht.covar\
     --out output/ozht.prs_rsq_${r2_thresh}_p_below_${pv_thresh}.reml
     
grep X_7 output/ozht.prs_rsq_${r2_thresh}_p_below_${pv_thresh}.reml.log &gt; output/ozht.prs_rsq_${r2_thresh}_p_below_${pv_thresh}.effect.txt</code></pre>
<p>Note X_7 is the variable for PRS score in this example. The order
depends on the order in the <code>--covar</code> and
<code>--qcovar</code> files.</p>
<p>Let’s calculate r2 and p value for this PRS score back in R</p>
<pre class="r"><code>phen &lt;- read.table(&quot;data/e2/ozht.phen&quot;, header = T)
all = merge(all, phen, by=c(&quot;IID&quot;, &quot;FID&quot;))
res &lt;- read.table(&quot;output/ozht.prs_rsq_0.1_p_below_5e-8.effect.txt&quot;, header = F)
betahat &lt;- res[1,2]
sehat &lt;- res[1,3]
r2 &lt;- (betahat/sd(all$ht, na.rm = T)*sd(all$SCORESUM, na.rm = T))**2
pval &lt;- (1-pt(q=betahat/sehat, df = 1894, lower.tail = T))*2
cat(&quot;r2: &quot;, r2, &quot;; p value: &quot;, pval)</code></pre>
<pre><code>r2:  0.009820507 ; p value:  0.0005102413</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /software/R-4.1.0-no-openblas-el7-x86_64/lib64/R/lib/libRblas.so
LAPACK: /software/R-4.1.0-no-openblas-el7-x86_64/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9       rstudioapi_0.13  knitr_1.42       magrittr_2.0.1  
 [5] workflowr_1.6.2  R6_2.5.0         rlang_1.1.0      fastmap_1.1.0   
 [9] fansi_0.5.0      stringr_1.4.0    tools_4.1.0      xfun_0.38       
[13] utf8_1.2.1       cli_3.6.1        git2r_0.28.0     jquerylib_0.1.4 
[17] htmltools_0.5.5  ellipsis_0.3.2   rprojroot_2.0.2  yaml_2.2.1      
[21] digest_0.6.27    tibble_3.1.2     lifecycle_1.0.3  crayon_1.5.2    
[25] later_1.2.0      sass_0.4.0       vctrs_0.3.8      promises_1.2.0.1
[29] fs_1.6.1         glue_1.4.2       cachem_1.0.5     evaluate_0.20   
[33] rmarkdown_2.21   stringi_1.6.2    bslib_0.4.2      compiler_4.1.0  
[37] pillar_1.6.1     jsonlite_1.7.2   httpuv_1.6.1     pkgconfig_2.0.3 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
